fvm/metrics.go

```diff
@@ -35,19 +35,28 @@ type metricsCollector struct {
 
 func (m metricsCollector) ProgramParsed(location common.Location, duration time.Duration) {
 	if m.MetricsCollector != nil {
-		m.parsed += duration
+		// only capture tx parsing time
+		if _, ok := location.(common.TransactionLocation); ok {
```

Comment:
These checks prevent re-reporting durations, the metrics collection is a bit counter-intuitive: The three functions (parsing, checking, interpretation) are not called in sequence, but in some cases as part of each other.

For example, when a program is checked, it may import another program, which is then in turn parsed + checked. Or during interpretation, a value may be loaded from storage, which leads to the program for it being loaded + checked.

We basically only measure the durations reported for the entry-point (the transaction), and not for child locations, because they might be already part of the duration for the entry-point.

Reply:
@ramtinms Could you please maybe add this / a variant of what I wrote above as a comment somewhere to the code? I think it will prevent future confusion, including by myself 😅 

Maybe we can improve the metric reporting in Cadence somehow, but I didn't find a good way to "subtract" child timings in parent operations.

Reply:
added as comments